Drop String constructors taking in a raw pointer and a length

cdumez opened the pull request:
#### c04fd9f6099bcea032ee39b16390a8850bdcdcbe
<pre>
Drop String constructors taking in a raw pointer and a length
<a href="https://bugs.webkit.org/show_bug.cgi?id=271888">https://bugs.webkit.org/show_bug.cgi?id=271888</a>

Reviewed by Darin Adler.

Drop String constructors taking in a raw pointer and a length, in favor of the
ones taking in a std::span.

* Source/JavaScriptCore/API/OpaqueJSString.h:
(OpaqueJSString::OpaqueJSString):
* Source/JavaScriptCore/API/tests/JSONParseTest.cpp:
(testJSONParse):
* Source/JavaScriptCore/dfg/DFGLazyJSValue.cpp:
(JSC::DFG::LazyJSValue::tryGetString const):
* Source/JavaScriptCore/parser/Lexer.cpp:
(JSC::Lexer&lt;CharacterType&gt;::parseCommentDirectiveValue):
* Source/JavaScriptCore/runtime/FuzzerPredictions.cpp:
(JSC::readFileIntoString):
* Source/JavaScriptCore/runtime/IntlCollator.cpp:
(JSC::IntlCollator::sortLocaleData):
(JSC::IntlCollator::checkICULocaleInvariants):
* Source/JavaScriptCore/runtime/IntlDateTimeFormat.cpp:
(JSC::IntlDateTimeFormat::localeData):
* Source/JavaScriptCore/runtime/IntlDisplayNames.cpp:
(JSC::IntlDisplayNames::of const):
* Source/JavaScriptCore/runtime/IntlDurationFormat.cpp:
(JSC::retrieveSeparator):
* Source/JavaScriptCore/runtime/IntlLocale.cpp:
(JSC::IntlLocale::language):
(JSC::IntlLocale::script):
(JSC::IntlLocale::region):
(JSC::IntlLocale::calendars):
(JSC::IntlLocale::collations):
(JSC::IntlLocale::timeZones):
* Source/JavaScriptCore/runtime/IntlNumberFormat.cpp:
(JSC::IntlNumberFormat::formatRange const):
* Source/JavaScriptCore/runtime/IntlObject.cpp:
(JSC::languageTagForLocaleID):
(JSC::intlAvailableCalendars):
(JSC::availableCollations):
(JSC::availableCurrencies):
(JSC::availableNumberingSystems):
(JSC::intlAvailableTimeZones):
* Source/JavaScriptCore/runtime/IntlPluralRules.cpp:
(JSC::IntlPluralRules::resolvedOptions const):
* Source/JavaScriptCore/runtime/JSImmutableButterfly.cpp:
(JSC::JSImmutableButterfly::createFromString):
* Source/JavaScriptCore/runtime/JSONObject.cpp:
(JSC::gap):
(JSC::FastStringifier&lt;CharType&gt;::result const):
* Source/JavaScriptCore/runtime/LiteralParser.cpp:
(JSC::LiteralParser&lt;CharType&gt;::makeJSString):
* Source/JavaScriptCore/runtime/NumberPrototype.cpp:
(JSC::toStringWithRadixInternal):
* Source/JavaScriptCore/tools/FunctionAllowlist.cpp:
(JSC::FunctionAllowlist::FunctionAllowlist):
* Source/JavaScriptCore/tools/FunctionOverrides.cpp:
(JSC::parseClause):
* Source/JavaScriptCore/wasm/WasmIndexOrName.cpp:
(JSC::Wasm::makeString):
* Source/WTF/wtf/URLHelpers.cpp:
(WTF::URLHelpers::mapHostName):
* Source/WTF/wtf/UUID.cpp:
(WTF::bootSessionUUIDString):
* Source/WTF/wtf/text/StringView.h:
(WTF::StringView::toString const):
* Source/WTF/wtf/text/WTFString.cpp:
(WTF::String::String):
(WTF::String::fromCodePoint):
* Source/WTF/wtf/text/WTFString.h:
(WTF::IntegerToStringConversionTrait&lt;String&gt;::flush):
(WTF::StringLiterals::operator _str):
* Source/WebCore/Modules/encryptedmedia/InitDataRegistry.cpp:
(WebCore::extractKeyIDsKeyids):
* Source/WebCore/Modules/websockets/WebSocketExtensionParser.cpp:
(WebCore::WebSocketExtensionParser::consumeToken):
* Source/WebCore/PAL/pal/text/TextCodecReplacement.cpp:
(PAL::TextCodecReplacement::decode):
* Source/WebCore/accessibility/AccessibilityMathMLElement.cpp:
(WebCore::AccessibilityMathMLElement::textUnderElement const):
* Source/WebCore/bindings/js/SerializedScriptValue.cpp:
(WebCore::CloneDeserializer::readString):
* Source/WebCore/crypto/SubtleCrypto.cpp:
(WebCore::SubtleCrypto::unwrapKey):
* Source/WebCore/css/CSSCounterStyle.cpp:
(WebCore::counterForSystemCJK):
(WebCore::CSSCounterStyle::counterForSystemEthiopicNumeric):
* Source/WebCore/css/process-css-properties.py:
(GenerateCSSPropertyNames):
* Source/WebCore/editing/Editing.cpp:
(WebCore::nonBreakingSpaceString):
* Source/WebCore/fileapi/FileReaderLoader.cpp:
(WebCore::FileReaderLoader::stringResult):
* Source/WebCore/html/FTPDirectoryDocument.cpp:
(WebCore::FTPDirectoryDocumentParser::appendEntry):
(WebCore::FTPDirectoryDocumentParser::loadDocumentTemplate):
* Source/WebCore/html/TypeAhead.cpp:
(WebCore::TypeAhead::handleEvent):
* Source/WebCore/html/parser/AtomHTMLToken.h:
(WebCore::AtomHTMLToken::AtomHTMLToken):
* Source/WebCore/html/parser/CSSPreloadScanner.cpp:
(WebCore::parseCSSStringOrURL):
* Source/WebCore/html/parser/HTMLDocumentParserFastPath.cpp:
(WebCore::HTMLFastPathParser::scanText):
* Source/WebCore/html/parser/HTMLEntityParser.cpp:
(WebCore::SegmentedStringSource::pushBackButKeep):
* Source/WebCore/html/parser/HTMLInputStream.h:
(WebCore::HTMLInputStream::markEndOfFile):
* Source/WebCore/html/shadow/DateTimeNumericFieldElement.cpp:
(WebCore::DateTimeNumericFieldElement::handleKeyboardEvent):
* Source/WebCore/html/track/VTTScanner.cpp:
(WebCore::VTTScanner::extractString):
* Source/WebCore/html/track/VTTScanner.h:
* Source/WebCore/html/track/WebVTTTokenizer.cpp:
(WebCore::WebVTTTokenizer::WebVTTTokenizer):
* Source/WebCore/inspector/InspectorStyleSheet.cpp:
(WebCore::fixUnparsedProperties):
* Source/WebCore/layout/formattingContexts/inline/display/InlineDisplayContentBuilder.cpp:
(WebCore::Layout::InlineDisplayContentBuilder::appendTextDisplayBox):
* Source/WebCore/loader/FTPDirectoryParser.cpp:
(WebCore::parseOneFTPLine):
* Source/WebCore/loader/FormSubmission.cpp:
(WebCore::appendMailtoPostFormDataToURL):
* Source/WebCore/loader/LinkHeader.cpp:
(WebCore::findURLBoundaries):
(WebCore::parseParameterValue):
* Source/WebCore/loader/ResourceCryptographicDigest.cpp:
(WebCore::parseCryptographicDigestImpl):
(WebCore::parseEncodedCryptographicDigestImpl):
* Source/WebCore/loader/appcache/ApplicationCacheManifestParser.cpp:
(WebCore::makeManifestURL):
* Source/WebCore/loader/appcache/ApplicationCacheStorage.cpp:
(WebCore::parseHeader):
* Source/WebCore/page/EventSource.cpp:
(WebCore::EventSource::dispatchMessageEvent):
* Source/WebCore/page/csp/ContentSecurityPolicy.cpp:
(WebCore::ContentSecurityPolicy::didReceiveHeader):
* Source/WebCore/page/csp/ContentSecurityPolicyDirectiveList.cpp:
(WebCore::ContentSecurityPolicyDirectiveList::parseDirective):
(WebCore::ContentSecurityPolicyDirectiveList::parseRequireTrustedTypesFor):
* Source/WebCore/page/csp/ContentSecurityPolicyMediaListDirective.cpp:
(WebCore::ContentSecurityPolicyMediaListDirective::parse):
* Source/WebCore/page/csp/ContentSecurityPolicySourceList.cpp:
(WebCore::ContentSecurityPolicySourceList::parse):
(WebCore::ContentSecurityPolicySourceList::parsePath):
(WebCore::ContentSecurityPolicySourceList::parseNonceSource):
* Source/WebCore/page/csp/ContentSecurityPolicyTrustedTypesDirective.cpp:
(WebCore::ContentSecurityPolicyTrustedTypesDirective::parse):
* Source/WebCore/platform/cocoa/KeyEventCocoa.mm:
(WebCore::keyForCharCode):
* Source/WebCore/platform/encryptedmedia/CDMUtilities.cpp:
(WebCore::CDMUtilities::parseJSONObject):
* Source/WebCore/platform/graphics/FontCascade.h:
(WebCore::FontCascade::widthOfSpaceString const):
* Source/WebCore/platform/graphics/StringTruncator.cpp:
(WebCore::truncateString):
* Source/WebCore/platform/graphics/TextRun.h:
(WebCore::TextRun::isolatedCopy const):
* Source/WebCore/platform/graphics/angle/GraphicsContextGLANGLE.cpp:
(WebCore::GraphicsContextGLANGLE::getActiveAttribImpl):
(WebCore::GraphicsContextGLANGLE::getActiveUniformImpl):
(WebCore::GraphicsContextGLANGLE::getProgramInfoLog):
(WebCore::GraphicsContextGLANGLE::getShaderInfoLog):
(WebCore::GraphicsContextGLANGLE::getActiveUniformBlockName):
(WebCore::GraphicsContextGLANGLE::getTransformFeedbackVarying):
(WebCore::GraphicsContextGLANGLE::getTranslatedShaderSourceANGLE):
* Source/WebCore/platform/graphics/avfoundation/CDMFairPlayStreaming.cpp:
(WebCore::extractSinfData):
(WebCore::CDMPrivateFairPlayStreaming::extractKeyIDsMpts):
* Source/WebCore/platform/graphics/avfoundation/MediaPlayerPrivateAVFoundation.cpp:
(WebCore::MediaPlayerPrivateAVFoundation::extractKeyURIKeyIDAndCertificateFromInitData):
* Source/WebCore/platform/graphics/avfoundation/objc/CDMInstanceFairPlayStreamingAVFObjC.mm:
(WebCore::parseJSONValue):
* Source/WebCore/platform/graphics/cocoa/FontPlatformDataCocoa.mm:
(WebCore::FontPlatformData::variationAxes const):
* Source/WebCore/platform/graphics/cocoa/SourceBufferParserWebM.cpp:
(WebCore::WebMParser::OnTrackEntry):
* Source/WebCore/platform/network/HTTPHeaderMap.cpp:
(WebCore::HTTPHeaderMap::set):
* Source/WebCore/platform/network/HTTPParsers.cpp:
(WebCore::trimInputSample):
(WebCore::parseHTTPHeader):
* Source/WebCore/platform/network/RFC8941.cpp:
(RFC8941::parseToken):
* Source/WebCore/platform/text/SegmentedString.cpp:
(WebCore::SegmentedString::advancePastSlowCase):
* Source/WebCore/rendering/RenderCombineText.cpp:
(WebCore::RenderCombineText::combineTextIfNeeded):
* Source/WebCore/rendering/RenderFileUploadControl.cpp:
(WebCore::RenderFileUploadControl::computeIntrinsicLogicalWidths const):
* Source/WebCore/rendering/RenderListMarker.cpp:
(WebCore::RenderListMarker::textRun const):
* Source/WebCore/rendering/RenderTextControl.cpp:
(WebCore::RenderTextControl::getAverageCharWidth):
* Source/WebCore/rendering/svg/SVGTextMetrics.cpp:
(WebCore::SVGTextMetrics::SVGTextMetrics):
* Source/WebCore/svg/SVGParserUtilities.cpp:
(WebCore::parseGlyphName):
(WebCore::parseKerningUnicodeString):
* Source/WebCore/svg/SVGStringList.cpp:
(WebCore::SVGStringList::parse):
* Source/WebCore/svg/SVGToOTFFontConversion.cpp:
(WebCore::codepointToString):
* Source/WebCore/svg/SVGViewSpec.cpp:
(WebCore::SVGViewSpec::parseViewSpec):
* Source/WebCore/testing/MockCDMFactory.cpp:
(WebCore::MockCDMInstanceSession::updateLicense):
* Source/WebKit/UIProcess/API/APIContentRuleListStore.cpp:
(API::getContentRuleListSourceFromMappedFile):
* Source/WebKit/UIProcess/Cocoa/WebPasteboardProxyCocoa.mm:
(WebKit::WebPasteboardProxy::testIPCSharedMemory):
* Source/WebKit/UIProcess/mac/WebViewImpl.mm:
(WebKit::commandNameForSelector):
* Source/WebKit/WebProcess/Network/webrtc/LibWebRTCResolver.cpp:
(WebKit::LibWebRTCResolver::start):
* Source/WebKitLegacy/mac/WebView/WebHTMLView.mm:
(commandNameForSelector):
* Tools/DumpRenderTree/TestRunner.cpp:
(TestRunner::runUIScript):
* Tools/TestRunnerShared/UIScriptContext/UIScriptContext.cpp:
(UIScriptContext::tryToCompleteUIScriptForCurrentParentCallback):
* Tools/TestWebKitAPI/Tests/WTF/HexNumber.cpp:
* Tools/TestWebKitAPI/Tests/WTF/NativePromise.cpp:
(TestWebKitAPI::PhotoProducer::takePhotoImpl const):
* Tools/TestWebKitAPI/Tests/WTF/StringBuilder.cpp:
(TestWebKitAPI::expectBuilderContent):
(TestWebKitAPI::TEST(StringBuilderTest, Append)):
(TestWebKitAPI::TEST(StringBuilderTest, VariadicAppend)):
* Tools/TestWebKitAPI/Tests/WTF/StringOperators.cpp:
(TestWebKitAPI::TEST(WTF, ConcatenateCharacterArrayAndEmptyString)):
* Tools/TestWebKitAPI/Tests/WTF/StringView.cpp:
(TestWebKitAPI::TEST(WTF, StringViewContainsOnlyASCII)):
* Tools/TestWebKitAPI/Tests/WTF/TextBreakIterator.cpp:
(TestWebKitAPI::makeUTF16):
* Tools/TestWebKitAPI/Tests/WTF/WTFString.cpp:
(TestWebKitAPI::TEST(WTF, StringUnicodeEqualUCharArray)):
* Tools/TestWebKitAPI/Tests/WTF/cocoa/URLExtras.mm:
(TestWebKitAPI::TEST(WTF_URLExtras, URLExtras_ParsingError)):
* Tools/TestWebKitAPI/Tests/WebCore/CBORReaderTest.cpp:
(TestWebKitAPI::TEST(CBORReaderTest, TestReadStringWithNUL)):
* Tools/TestWebKitAPI/Tests/WebCore/FileMonitor.cpp:
(TestWebKitAPI::readContentsOfFile):
* Tools/TestWebKitAPI/Tests/WebCore/SharedBuffer.cpp:
(TestWebKitAPI::TEST_F):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/Download.mm:
(longString):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/EventAttribution.mm:
(TestWebKitAPI::signUnlinkableTokenAndSendSecretToken):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/ServiceWorkerBasic.mm:
((ServiceWorker, ExtensionServiceWorkerDisableCORS)):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WKWebExtensionAPIWebNavigation.mm:
(TestWebKitAPI::longString):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WebPushDaemon.mm:
* Tools/TestWebKitAPI/cocoa/HTTPServer.mm:
(TestWebKitAPI::HTTPServer::parsePath):
(TestWebKitAPI::HTTPServer::parseBody):
* Tools/WebKitTestRunner/TestController.cpp:
(WTR::TestController::createTestURL):

Canonical link: <a href="https://commits.webkit.org/276859@main">https://commits.webkit.org/276859@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/e949a4fad012bbdafde28cc2d659e34fdf206e03

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/45979 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/25108 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/48564 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/48650 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/42019 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/48286 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/29418 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/22505 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/37600 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/46557 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/22193 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/39662 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/18790 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/19590 "Passed tests") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/10/builds/40758 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/4023 "Built successfully") | 
| [  ~~🛠 🧪 jsc~~](https://ews-build.webkit.org/#/builders/20/builds/39212 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/42286 "Passed tests") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/41109 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/2/builds/50436 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/45453 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/20975 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/17452 "Passed tests") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/44759 "layout-tests (failure)") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/22275 "Built successfully") | | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/43651 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/22634 "Built successfully") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/52604 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/6403 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/21969 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/10747 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

darinadler commented on Source/JavaScriptCore/dfg/DFGLazyJSValue.cpp:
```diff
@@ -131,7 +131,7 @@ String LazyJSValue::tryGetString(Graph& graph) const
 
     case SingleCharacterString:
-        return String(&u.character, 1);
+        return String(span(u.character));
```
Do we need `String()` here?

cdumez replied:
We do not.

darinadler commented on Source/JavaScriptCore/runtime/FuzzerPredictions.cpp:
```diff
@@ -43,7 +43,7 @@ static String readFileIntoString(const char* fileName)
     fclose(file);
     RELEASE_ASSERT(readSize == buffer.size());
-    return String(buffer.data(), buffer.size());
+    return String(buffer.span());
```
Do we need `String()` here?

‘I also think `String::createUninitialized` would allow us to do this without two copies. And there should be a helper function version of this, not written out here.

darinadler commented on Source/JavaScriptCore/runtime/IntlDisplayNames.cpp:
```diff
@@ -340,7 +340,7 @@ JSValue IntlDisplayNames::of(JSGlobalObject* globalObject, JSValue codeValue) co
         // We should return undefined if fallback is "none". Otherwise, we should return input value.
         if (status == U_ILLEGAL_ARGUMENT_ERROR)
-            return (m_fallback == Fallback::None) ? jsUndefined() : jsString(vm, String(canonicalCode.data(), canonicalCode.length()));
+            return (m_fallback == Fallback::None) ? jsUndefined() : jsString(vm, String(canonicalCode.span()));
```
Maybe we can omit `String()` here?

darinadler commented on Source/JavaScriptCore/runtime/IntlLocale.cpp:
```diff
@@ -457,7 +457,7 @@ const String& IntlLocale::language()
         auto status = callBufferProducingFunction(uloc_getLanguage, m_localeID.data(), buffer);
         ASSERT_UNUSED(status, U_SUCCESS(status));
-        m_language = String(buffer.data(), buffer.size());
+        m_language = String(buffer.span());
```
Can we omit `String()` here?

darinadler commented on Source/JavaScriptCore/runtime/IntlLocale.cpp:
```diff
@@ -469,7 +469,7 @@ const String& IntlLocale::script()
         auto status = callBufferProducingFunction(uloc_getScript, m_localeID.data(), buffer);
         ASSERT_UNUSED(status, U_SUCCESS(status));
-        m_script = String(buffer.data(), buffer.size());
+        m_script = String(buffer.span());
```
Can we omit `String()` here?

darinadler commented on Source/JavaScriptCore/runtime/IntlLocale.cpp:
```diff
@@ -481,7 +481,7 @@ const String& IntlLocale::region()
         auto status = callBufferProducingFunction(uloc_getCountry, m_localeID.data(), buffer);
         ASSERT_UNUSED(status, U_SUCCESS(status));
-        m_region = String(buffer.data(), buffer.size());
+        m_region = String(buffer.span());
```
Can we omit `String()` here?

darinadler commented on Source/JavaScriptCore/runtime/IntlObject.cpp:
```diff
@@ -405,7 +405,7 @@ String languageTagForLocaleID(const char* localeID, bool isImmortal)
         if (isImmortal)
             return StringImpl::createStaticStringImpl(buffer.data(), buffer.size());
-        return String(buffer.data(), buffer.size());
+        return String(buffer.span());
```
Can we omit `String()` here?

darinadler commented on Source/JavaScriptCore/runtime/JSImmutableButterfly.cpp:
```diff
@@ -222,11 +222,11 @@ JSImmutableButterfly* JSImmutableButterfly::createFromString(JSGlobalObject* glo
                 characters[index + 1],
             };
-            value = jsNontrivialString(vm, String(string, 2));
+            value = jsNontrivialString(vm, String(std::span { string }));
```
Can we omit `String()` here?

cdumez replied:
Best I could do was: `value = jsNontrivialString(vm, String(string));`

darinadler commented on Source/JavaScriptCore/wasm/WasmIndexOrName.cpp:
```diff
@@ -56,7 +56,7 @@ String makeString(const IndexOrName& ion)
     if (ion.isEmpty())
         return "wasm-stub"_s;
-    const String moduleName = ion.nameSection()->moduleName.size() ? String(ion.nameSection()->moduleName.data(), ion.nameSection()->moduleName.size()) : String(ion.nameSection()->moduleHash.data(), ion.nameSection()->moduleHash.size());
+    const String moduleName = ion.nameSection()->moduleName.size() ? String(ion.nameSection()->moduleName.span()) : String(ion.nameSection()->moduleHash.span());
```
I think we can use the `span` here rather than constructing a `String`.

darinadler commented on Source/WTF/wtf/text/StringView.h:
```diff
@@ -652,8 +652,8 @@ inline StringView::UpconvertedCharactersWithSize<N>::UpconvertedCharactersWithSi
     if (is8Bit())
-        return String(characters8(), m_length);
-    return String(characters16(), m_length);
+        return String(span8());
```
Can omit `String()` here.

darinadler commented on Source/WTF/wtf/text/StringView.h:
```diff
@@ -652,8 +652,8 @@ inline StringView::UpconvertedCharactersWithSize<N>::UpconvertedCharactersWithSi
-        return String(characters8(), m_length);
-    return String(characters16(), m_length);
+        return String(span8());
+    return String(span16());
```
Can omit `String()` here.

darinadler commented on Source/WebCore/Modules/mediastream/gstreamer/GStreamerWebRTCUtils.cpp:
```diff
@@ -339,7 +339,7 @@ static String x509Serialize(X509* x509)
         return { };
 
-    return String(buffer.data(), length);
+    return String(buffer.subspan(0, length));
```
Can omit `String()` here.

darinadler commented on Source/WebCore/Modules/mediastream/gstreamer/GStreamerWebRTCUtils.cpp:
```diff
@@ -357,7 +357,7 @@ static String privateKeySerialize(EVP_PKEY* privateKey)
         return { };
 
-    return String(buffer.data(), length);
+    return String(buffer.subspan(0, length));
```
Can omit `String()` here.

darinadler commented on Source/WebCore/PAL/pal/text/TextCodecReplacement.cpp:
```diff
@@ -56,7 +56,7 @@ String TextCodecReplacement::decode(std::span<const uint8_t>, bool, bool, bool&
         return emptyString();
     m_sentEOF = true;
-    return String { &replacementCharacter, 1 };
+    return String { span(replacementCharacter) };
```
Can omit `String{}` here.

darinadler commented on Source/WebCore/html/FTPDirectoryDocument.cpp:
```diff
@@ -114,7 +114,7 @@ void FTPDirectoryDocumentParser::appendEntry(String&& filename, String&& size, S
 
     Ref typeElement = HTMLTableCellElement::create(tdTag, document);
-    typeElement->appendChild(Text::create(document, String(&noBreakSpace, 1)));
+    typeElement->appendChild(Text::create(document, String(span(noBreakSpace))));
```
Can omit `String()` here?

darinadler commented on Source/WebCore/html/FTPDirectoryDocument.cpp:
```diff
@@ -298,7 +298,7 @@ bool FTPDirectoryDocumentParser::loadDocumentTemplate()
     }
 
-    HTMLDocumentParser::insert(String(templateDocumentData.get()->data(), templateDocumentData.get()->size()));
+    HTMLDocumentParser::insert(String(templateDocumentData.get()->span()));
```
Can omit `String()` here?

darinadler commented on Source/WebCore/html/TypeAhead.cpp:
```diff
@@ -76,7 +76,7 @@ int TypeAhead::handleEvent(KeyboardEvent* event, MatchModeFlags matchMode)
         // The user is likely trying to cycle through all the items starting
         // with this character, so just search on the character.
-        prefix = String(&c, 1);
+        prefix = String(span(c));
```
Can omit `String()` here.

darinadler commented on Source/WebCore/html/parser/AtomHTMLToken.h:
```diff
@@ -270,7 +270,7 @@ inline AtomHTMLToken::AtomHTMLToken(HTMLToken& token)
         else
-            m_data = String(token.comment().data(), token.comment().size());
+            m_data = String(token.comment().span());
         return;
```
Can omit `String()` here.

darinadler commented on Source/WebCore/html/parser/HTMLEntityParser.cpp:
```diff
@@ -160,7 +160,7 @@ void SegmentedStringSource::pushBackButKeep(unsigned keepCount)
     ASSERT(keepCount < m_consumedCharacters.size());
     unsigned length = m_consumedCharacters.size() - keepCount;
-    m_source.pushBack(String { m_consumedCharacters.data() + keepCount, length });
+    m_source.pushBack(String { m_consumedCharacters.subspan(keepCount, length) });
```
Can omit `String{}` here.

darinadler commented on Source/WebCore/html/parser/HTMLInputStream.h:
```diff
@@ -72,7 +72,7 @@ class HTMLInputStream {
     void markEndOfFile()
     {
-        m_last->append(String { &kEndOfFileMarker, 1 });
+        m_last->append(String { span(kEndOfFileMarker) });
```
Can omit `String{}` here.

darinadler commented on Source/WebCore/html/track/VTTScanner.h:
```diff
@@ -131,6 +131,7 @@ class VTTScanner {
 protected:
     Position position() const { return m_data.characters8; }
     Position end() const { return m_end.characters8; }
+
```
Revert this blank line change?

darinadler commented on Source/WebCore/html/track/WebVTTTokenizer.cpp:
```diff
@@ -86,7 +86,7 @@ WebVTTTokenizer::WebVTTTokenizer(const String& input)
     ASSERT(!m_input.isClosed());
-    m_input.append(String { &kEndOfFileMarker, 1 });
+    m_input.append(String { span(kEndOfFileMarker) });
     m_input.close();
```
Can omit `String{}` here.

darinadler commented on Source/WebCore/loader/FormSubmission.cpp:
```diff
@@ -76,7 +76,7 @@ static void appendMailtoPostFormDataToURL(URL& url, const FormData& data, const
     Vector<uint8_t> bodyData(std::span<const char>("body=", static_cast<size_t>(5)));
     FormDataBuilder::encodeStringAsFormData(bodyData, body.utf8());
-    body = makeStringByReplacingAll(String(bodyData.data(), bodyData.size()), '+', "%20"_s);
+    body = makeStringByReplacingAll(String(bodyData.span()), '+', "%20"_s);
```
Can omit `String()` here?

darinadler commented on Source/WebCore/platform/cocoa/KeyEventCocoa.mm:
```diff
@@ -190,7 +190,7 @@ String keyForCharCode(unichar charCode)
         return "Unidentified"_s;
     default:
-        return String(reinterpret_cast<UChar*>(&charCode), 1);
+        return String(span(*reinterpret_cast<const UChar*>(&charCode)));
```
Can omit `String()` here.

darinadler commented on Source/WebCore/platform/graphics/FontCascade.h:
```diff
@@ -144,7 +144,7 @@ class FontCascade : public CanMakeWeakPtr<FontCascade>, public CanMakeCheckedPtr
     float widthOfSpaceString() const
     {
-        return width(TextRun { String { &space, 1 } });
+        return width(TextRun { String({ &space, 1 }) });
```
Can we make this cleaner with the single character `span` function?

darinadler commented on Source/WebCore/platform/graphics/cocoa/FontPlatformDataCocoa.mm:
```diff
@@ -63,7 +63,7 @@
     return WTF::map(defaultVariationValues(platformFont, shouldLocalizeAxisNames), [](auto&& entry) {
         auto& [tag, values] = entry;
-        return FontPlatformData::FontVariationAxis { values.axisName, String(tag.data(), tag.size()), values.defaultValue, values.minimumValue, values.maximumValue };
+        return FontPlatformData::FontVariationAxis { values.axisName, String(std::span { tag }), values.defaultValue, values.minimumValue, values.maximumValue };
```
Can omit `String()` here.

darinadler commented on Source/WebCore/platform/graphics/freetype/FontPlatformDataFreeType.cpp:
```diff
@@ -218,7 +218,7 @@ Vector<FontPlatformData::FontVariationAxis> FontPlatformData::variationAxes(Shou
     return WTF::map(defaultVariationValues(ftFace, shouldLocalizeAxisNames), [](auto&& entry) {
         auto& [tag, values] = entry;
-        return FontPlatformData::FontVariationAxis { values.axisName, String(tag.data(), tag.size()), values.defaultValue, values.minimumValue, values.maximumValue };
+        return FontPlatformData::FontVariationAxis { values.axisName, String(tag), values.defaultValue, values.minimumValue, values.maximumValue };
```
Can omit `String()` here.

darinadler commented on Source/WebCore/platform/graphics/skia/FontPlatformDataSkia.cpp:
```diff
@@ -150,7 +150,7 @@ Vector<FontPlatformData::FontVariationAxis> FontPlatformData::variationAxes(Shou
     return WTF::map(defaultFontVariationValues(*typeface), [](auto&& entry) {
         auto& [tag, values] = entry;
-        return FontPlatformData::FontVariationAxis { values.axisName, String(tag.data(), tag.size()), values.defaultValue, values.minimumValue, values.maximumValue };
+        return FontPlatformData::FontVariationAxis { values.axisName, String(tag), values.defaultValue, values.minimumValue, values.maximumValue };
```
Can omit `String()` here.

darinadler commented on Source/WebCore/platform/network/curl/CurlRequest.cpp:
```diff
@@ -643,7 +643,7 @@ size_t CurlRequest::willSendDataCallback(char* ptr, size_t blockSize, size_t num
 {
-    return static_cast<CurlRequest*>(userData)->didReceiveHeader(String(ptr, blockSize * numberOfBlocks));
+    return static_cast<CurlRequest*>(userData)->didReceiveHeader(String({ ptr, blockSize * numberOfBlocks }));
 }
```
Can omit `String()` here.

darinadler commented on Source/WebCore/platform/win/KeyEventWin.cpp:
```diff
@@ -219,7 +219,7 @@ static int windowsKeycodeWithLocation(WPARAM keycode, LPARAM keyData)
 static inline String singleCharacterString(UChar c)
 {
-    return String(&c, 1);
+    return String(span(c));
```
Can omit `String()` here.

darinadler commented on Source/WebDriver/Session.cpp:
```diff
@@ -2294,7 +2294,7 @@ void Session::elementSendKeys(const String& elementID, const String& text, Funct
                             }
                         } else
-                            interaction.text = String(&key, 1);
+                            interaction.text = String(span(key));
```
Can omit `String()` here.

darinadler commented on Source/WebKit/Shared/win/WebEventFactory.cpp:
```diff
@@ -215,7 +215,7 @@ static String textFromEvent(WPARAM wparam, WebEventType type)
 
     UChar c = static_cast<UChar>(wparam);
-    return String(&c, 1);
+    return String(span(c));
```
Can omit `String()` here.

darinadler commented on Source/WebKit/Shared/win/WebEventFactory.cpp:
```diff
@@ -224,7 +224,7 @@ static String unmodifiedTextFromEvent(WPARAM wparam, WebEventType type)
 
     UChar c = static_cast<UChar>(wparam);
-    return String(&c, 1);
+    return String(span(c));
```
Can omit `String()` here.

darinadler commented on Tools/TestWebKitAPI/Tests/WebKitCocoa/Download.mm:
```diff
@@ -1172,7 +1172,7 @@ HTTPServer server([receivedFirstConnection = false] (Connection connection) muta
 {
     Vector<LChar> vector(length, c);
-    return String(vector.data(), length);
+    return String(vector.span());
```
Can omit `String()` here.

darinadler commented on Tools/TestWebKitAPI/Tests/WebKitCocoa/WKWebExtensionAPIWebNavigation.mm:
```diff
@@ -268,7 +268,7 @@
 {
     Vector<LChar> vector(length, c);
-    return String(vector.data(), length);
+    return String(vector.span());
```
Can omit `String()` here.

darinadler commented on Tools/TestWebKitAPI/cocoa/HTTPServer.mm:
```diff
@@ -375,15 +375,15 @@ static void appendUTF8ToVector(Vector<uint8_t>& vector, const String& string)
     ASSERT_WITH_MESSAGE(pathPrefixLength, "HTTPServer assumes request is GET or POST");
     size_t pathLength = pathEnd - request.data() - pathPrefixLength;
-    return String(request.data() + pathPrefixLength, pathLength);
+    return String(request.subspan(pathPrefixLength, pathLength));
```
Can omit `String()` here.

darinadler commented on Tools/TestWebKitAPI/cocoa/HTTPServer.mm:
```diff
@@ -375,15 +375,15 @@ static void appendUTF8ToVector(Vector<uint8_t>& vector, const String& string)
     const char* headerEnd = strnstr(request.data(), headerEndBytes, request.size()) + strlen(headerEndBytes);
     size_t headerLength = headerEnd - request.data();
-    return String(headerEnd, request.size() - headerLength);
+    return String(request.subspan(headerLength));
```
Can omit `String()` here.

cdumez committed:
Drop String constructors taking in a raw pointer and a length
https://bugs.webkit.org/show_bug.cgi?id=271888

Reviewed by Darin Adler.

Drop String constructors taking in a raw pointer and a length, in favor of the
ones taking in a std::span.

* Source/JavaScriptCore/API/OpaqueJSString.h:
(OpaqueJSString::OpaqueJSString):
* Source/JavaScriptCore/API/tests/JSONParseTest.cpp:
(testJSONParse):
* Source/JavaScriptCore/dfg/DFGLazyJSValue.cpp:
(JSC::DFG::LazyJSValue::tryGetString const):
* Source/JavaScriptCore/parser/Lexer.cpp:
(JSC::Lexer<CharacterType>::parseCommentDirectiveValue):
* Source/JavaScriptCore/runtime/FuzzerPredictions.cpp:
(JSC::readFileIntoString):
* Source/JavaScriptCore/runtime/IntlCollator.cpp:
(JSC::IntlCollator::sortLocaleData):
(JSC::IntlCollator::checkICULocaleInvariants):
* Source/JavaScriptCore/runtime/IntlDateTimeFormat.cpp:
(JSC::IntlDateTimeFormat::localeData):
* Source/JavaScriptCore/runtime/IntlDisplayNames.cpp:
(JSC::IntlDisplayNames::of const):
* Source/JavaScriptCore/runtime/IntlDurationFormat.cpp:
(JSC::retrieveSeparator):
* Source/JavaScriptCore/runtime/IntlLocale.cpp:
(JSC::IntlLocale::language):
(JSC::IntlLocale::script):
(JSC::IntlLocale::region):
(JSC::IntlLocale::calendars):
(JSC::IntlLocale::collations):
(JSC::IntlLocale::timeZones):
* Source/JavaScriptCore/runtime/IntlNumberFormat.cpp:
(JSC::IntlNumberFormat::formatRange const):
* Source/JavaScriptCore/runtime/IntlObject.cpp:
(JSC::languageTagForLocaleID):
(JSC::intlAvailableCalendars):
(JSC::availableCollations):
(JSC::availableCurrencies):
(JSC::availableNumberingSystems):
(JSC::intlAvailableTimeZones):
* Source/JavaScriptCore/runtime/IntlPluralRules.cpp:
(JSC::IntlPluralRules::resolvedOptions const):
* Source/JavaScriptCore/runtime/JSImmutableButterfly.cpp:
(JSC::JSImmutableButterfly::createFromString):
* Source/JavaScriptCore/runtime/JSONObject.cpp:
(JSC::gap):
(JSC::FastStringifier<CharType>::result const):
* Source/JavaScriptCore/runtime/LiteralParser.cpp:
(JSC::LiteralParser<CharType>::makeJSString):
* Source/JavaScriptCore/runtime/NumberPrototype.cpp:
(JSC::toStringWithRadixInternal):
* Source/JavaScriptCore/tools/FunctionAllowlist.cpp:
(JSC::FunctionAllowlist::FunctionAllowlist):
* Source/JavaScriptCore/tools/FunctionOverrides.cpp:
(JSC::parseClause):
* Source/JavaScriptCore/wasm/WasmIndexOrName.cpp:
(JSC::Wasm::makeString):
* Source/WTF/wtf/URLHelpers.cpp:
(WTF::URLHelpers::mapHostName):
* Source/WTF/wtf/UUID.cpp:
(WTF::bootSessionUUIDString):
* Source/WTF/wtf/text/StringView.h:
(WTF::StringView::toString const):
* Source/WTF/wtf/text/WTFString.cpp:
(WTF::String::String):
(WTF::String::fromCodePoint):
* Source/WTF/wtf/text/WTFString.h:
(WTF::IntegerToStringConversionTrait<String>::flush):
(WTF::StringLiterals::operator _str):
* Source/WebCore/Modules/encryptedmedia/InitDataRegistry.cpp:
(WebCore::extractKeyIDsKeyids):
* Source/WebCore/Modules/websockets/WebSocketExtensionParser.cpp:
(WebCore::WebSocketExtensionParser::consumeToken):
* Source/WebCore/PAL/pal/text/TextCodecReplacement.cpp:
(PAL::TextCodecReplacement::decode):
* Source/WebCore/accessibility/AccessibilityMathMLElement.cpp:
(WebCore::AccessibilityMathMLElement::textUnderElement const):
* Source/WebCore/bindings/js/SerializedScriptValue.cpp:
(WebCore::CloneDeserializer::readString):
* Source/WebCore/crypto/SubtleCrypto.cpp:
(WebCore::SubtleCrypto::unwrapKey):
* Source/WebCore/css/CSSCounterStyle.cpp:
(WebCore::counterForSystemCJK):
(WebCore::CSSCounterStyle::counterForSystemEthiopicNumeric):
* Source/WebCore/css/process-css-properties.py:
(GenerateCSSPropertyNames):
* Source/WebCore/editing/Editing.cpp:
(WebCore::nonBreakingSpaceString):
* Source/WebCore/fileapi/FileReaderLoader.cpp:
(WebCore::FileReaderLoader::stringResult):
* Source/WebCore/html/FTPDirectoryDocument.cpp:
(WebCore::FTPDirectoryDocumentParser::appendEntry):
(WebCore::FTPDirectoryDocumentParser::loadDocumentTemplate):
* Source/WebCore/html/TypeAhead.cpp:
(WebCore::TypeAhead::handleEvent):
* Source/WebCore/html/parser/AtomHTMLToken.h:
(WebCore::AtomHTMLToken::AtomHTMLToken):
* Source/WebCore/html/parser/CSSPreloadScanner.cpp:
(WebCore::parseCSSStringOrURL):
* Source/WebCore/html/parser/HTMLDocumentParserFastPath.cpp:
(WebCore::HTMLFastPathParser::scanText):
* Source/WebCore/html/parser/HTMLEntityParser.cpp:
(WebCore::SegmentedStringSource::pushBackButKeep):
* Source/WebCore/html/parser/HTMLInputStream.h:
(WebCore::HTMLInputStream::markEndOfFile):
* Source/WebCore/html/shadow/DateTimeNumericFieldElement.cpp:
(WebCore::DateTimeNumericFieldElement::handleKeyboardEvent):
* Source/WebCore/html/track/VTTScanner.cpp:
(WebCore::VTTScanner::extractString):
* Source/WebCore/html/track/VTTScanner.h:
* Source/WebCore/html/track/WebVTTTokenizer.cpp:
(WebCore::WebVTTTokenizer::WebVTTTokenizer):
* Source/WebCore/inspector/InspectorStyleSheet.cpp:
(WebCore::fixUnparsedProperties):
* Source/WebCore/layout/formattingContexts/inline/display/InlineDisplayContentBuilder.cpp:
(WebCore::Layout::InlineDisplayContentBuilder::appendTextDisplayBox):
* Source/WebCore/loader/FTPDirectoryParser.cpp:
(WebCore::parseOneFTPLine):
* Source/WebCore/loader/FormSubmission.cpp:
(WebCore::appendMailtoPostFormDataToURL):
* Source/WebCore/loader/LinkHeader.cpp:
(WebCore::findURLBoundaries):
(WebCore::parseParameterValue):
* Source/WebCore/loader/ResourceCryptographicDigest.cpp:
(WebCore::parseCryptographicDigestImpl):
(WebCore::parseEncodedCryptographicDigestImpl):
* Source/WebCore/loader/appcache/ApplicationCacheManifestParser.cpp:
(WebCore::makeManifestURL):
* Source/WebCore/loader/appcache/ApplicationCacheStorage.cpp:
(WebCore::parseHeader):
* Source/WebCore/page/EventSource.cpp:
(WebCore::EventSource::dispatchMessageEvent):
* Source/WebCore/page/csp/ContentSecurityPolicy.cpp:
(WebCore::ContentSecurityPolicy::didReceiveHeader):
* Source/WebCore/page/csp/ContentSecurityPolicyDirectiveList.cpp:
(WebCore::ContentSecurityPolicyDirectiveList::parseDirective):
(WebCore::ContentSecurityPolicyDirectiveList::parseRequireTrustedTypesFor):
* Source/WebCore/page/csp/ContentSecurityPolicyMediaListDirective.cpp:
(WebCore::ContentSecurityPolicyMediaListDirective::parse):
* Source/WebCore/page/csp/ContentSecurityPolicySourceList.cpp:
(WebCore::ContentSecurityPolicySourceList::parse):
(WebCore::ContentSecurityPolicySourceList::parsePath):
(WebCore::ContentSecurityPolicySourceList::parseNonceSource):
* Source/WebCore/page/csp/ContentSecurityPolicyTrustedTypesDirective.cpp:
(WebCore::ContentSecurityPolicyTrustedTypesDirective::parse):
* Source/WebCore/platform/cocoa/KeyEventCocoa.mm:
(WebCore::keyForCharCode):
* Source/WebCore/platform/encryptedmedia/CDMUtilities.cpp:
(WebCore::CDMUtilities::parseJSONObject):
* Source/WebCore/platform/graphics/FontCascade.h:
(WebCore::FontCascade::widthOfSpaceString const):
* Source/WebCore/platform/graphics/StringTruncator.cpp:
(WebCore::truncateString):
* Source/WebCore/platform/graphics/TextRun.h:
(WebCore::TextRun::isolatedCopy const):
* Source/WebCore/platform/graphics/angle/GraphicsContextGLANGLE.cpp:
(WebCore::GraphicsContextGLANGLE::getActiveAttribImpl):
(WebCore::GraphicsContextGLANGLE::getActiveUniformImpl):
(WebCore::GraphicsContextGLANGLE::getProgramInfoLog):
(WebCore::GraphicsContextGLANGLE::getShaderInfoLog):
(WebCore::GraphicsContextGLANGLE::getActiveUniformBlockName):
(WebCore::GraphicsContextGLANGLE::getTransformFeedbackVarying):
(WebCore::GraphicsContextGLANGLE::getTranslatedShaderSourceANGLE):
* Source/WebCore/platform/graphics/avfoundation/CDMFairPlayStreaming.cpp:
(WebCore::extractSinfData):
(WebCore::CDMPrivateFairPlayStreaming::extractKeyIDsMpts):
* Source/WebCore/platform/graphics/avfoundation/MediaPlayerPrivateAVFoundation.cpp:
(WebCore::MediaPlayerPrivateAVFoundation::extractKeyURIKeyIDAndCertificateFromInitData):
* Source/WebCore/platform/graphics/avfoundation/objc/CDMInstanceFairPlayStreamingAVFObjC.mm:
(WebCore::parseJSONValue):
* Source/WebCore/platform/graphics/cocoa/FontPlatformDataCocoa.mm:
(WebCore::FontPlatformData::variationAxes const):
* Source/WebCore/platform/graphics/cocoa/SourceBufferParserWebM.cpp:
(WebCore::WebMParser::OnTrackEntry):
* Source/WebCore/platform/network/HTTPHeaderMap.cpp:
(WebCore::HTTPHeaderMap::set):
* Source/WebCore/platform/network/HTTPParsers.cpp:
(WebCore::trimInputSample):
(WebCore::parseHTTPHeader):
* Source/WebCore/platform/network/RFC8941.cpp:
(RFC8941::parseToken):
* Source/WebCore/platform/text/SegmentedString.cpp:
(WebCore::SegmentedString::advancePastSlowCase):
* Source/WebCore/rendering/RenderCombineText.cpp:
(WebCore::RenderCombineText::combineTextIfNeeded):
* Source/WebCore/rendering/RenderFileUploadControl.cpp:
(WebCore::RenderFileUploadControl::computeIntrinsicLogicalWidths const):
* Source/WebCore/rendering/RenderListMarker.cpp:
(WebCore::RenderListMarker::textRun const):
* Source/WebCore/rendering/RenderTextControl.cpp:
(WebCore::RenderTextControl::getAverageCharWidth):
* Source/WebCore/rendering/svg/SVGTextMetrics.cpp:
(WebCore::SVGTextMetrics::SVGTextMetrics):
* Source/WebCore/svg/SVGParserUtilities.cpp:
(WebCore::parseGlyphName):
(WebCore::parseKerningUnicodeString):
* Source/WebCore/svg/SVGStringList.cpp:
(WebCore::SVGStringList::parse):
* Source/WebCore/svg/SVGToOTFFontConversion.cpp:
(WebCore::codepointToString):
* Source/WebCore/svg/SVGViewSpec.cpp:
(WebCore::SVGViewSpec::parseViewSpec):
* Source/WebCore/testing/MockCDMFactory.cpp:
(WebCore::MockCDMInstanceSession::updateLicense):
* Source/WebKit/UIProcess/API/APIContentRuleListStore.cpp:
(API::getContentRuleListSourceFromMappedFile):
* Source/WebKit/UIProcess/Cocoa/WebPasteboardProxyCocoa.mm:
(WebKit::WebPasteboardProxy::testIPCSharedMemory):
* Source/WebKit/UIProcess/mac/WebViewImpl.mm:
(WebKit::commandNameForSelector):
* Source/WebKit/WebProcess/Network/webrtc/LibWebRTCResolver.cpp:
(WebKit::LibWebRTCResolver::start):
* Source/WebKitLegacy/mac/WebView/WebHTMLView.mm:
(commandNameForSelector):
* Tools/DumpRenderTree/TestRunner.cpp:
(TestRunner::runUIScript):
* Tools/TestRunnerShared/UIScriptContext/UIScriptContext.cpp:
(UIScriptContext::tryToCompleteUIScriptForCurrentParentCallback):
* Tools/TestWebKitAPI/Tests/WTF/HexNumber.cpp:
* Tools/TestWebKitAPI/Tests/WTF/NativePromise.cpp:
(TestWebKitAPI::PhotoProducer::takePhotoImpl const):
* Tools/TestWebKitAPI/Tests/WTF/StringBuilder.cpp:
(TestWebKitAPI::expectBuilderContent):
(TestWebKitAPI::TEST(StringBuilderTest, Append)):
(TestWebKitAPI::TEST(StringBuilderTest, VariadicAppend)):
* Tools/TestWebKitAPI/Tests/WTF/StringOperators.cpp:
(TestWebKitAPI::TEST(WTF, ConcatenateCharacterArrayAndEmptyString)):
* Tools/TestWebKitAPI/Tests/WTF/StringView.cpp:
(TestWebKitAPI::TEST(WTF, StringViewContainsOnlyASCII)):
* Tools/TestWebKitAPI/Tests/WTF/TextBreakIterator.cpp:
(TestWebKitAPI::makeUTF16):
* Tools/TestWebKitAPI/Tests/WTF/WTFString.cpp:
(TestWebKitAPI::TEST(WTF, StringUnicodeEqualUCharArray)):
* Tools/TestWebKitAPI/Tests/WTF/cocoa/URLExtras.mm:
(TestWebKitAPI::TEST(WTF_URLExtras, URLExtras_ParsingError)):
* Tools/TestWebKitAPI/Tests/WebCore/CBORReaderTest.cpp:
(TestWebKitAPI::TEST(CBORReaderTest, TestReadStringWithNUL)):
* Tools/TestWebKitAPI/Tests/WebCore/FileMonitor.cpp:
(TestWebKitAPI::readContentsOfFile):
* Tools/TestWebKitAPI/Tests/WebCore/SharedBuffer.cpp:
(TestWebKitAPI::TEST_F):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/Download.mm:
(longString):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/EventAttribution.mm:
(TestWebKitAPI::signUnlinkableTokenAndSendSecretToken):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/ServiceWorkerBasic.mm:
((ServiceWorker, ExtensionServiceWorkerDisableCORS)):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WKWebExtensionAPIWebNavigation.mm:
(TestWebKitAPI::longString):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WebPushDaemon.mm:
* Tools/TestWebKitAPI/cocoa/HTTPServer.mm:
(TestWebKitAPI::HTTPServer::parsePath):
(TestWebKitAPI::HTTPServer::parseBody):
* Tools/WebKitTestRunner/TestController.cpp:
(WTR::TestController::createTestURL):

Canonical link: https://commits.webkit.org/276859@main
